kafka-client/src/main/java/dev/responsive/kafka/store/ResponsivePartitionedStore.java

```diff
@@ -130,22 +143,53 @@ public void init(final StateStoreContext context, final StateStore root) {
           new BytesKeySpec(),
           config
       );
-      buffer.init();
 
-      open = true;
+      if (isActive()) {
+        initializeBuffer();
+      }
 
-      context.register(root, buffer);
-      final long offset = buffer.offset();
-      registration = new ResponsiveStoreRegistration(
-          name.kafkaName(),
-          topicPartition,
-          offset == -1 ? 0 : offset,
-          buffer::flush
+      open = true;
+      context.register(
+          root,
+          new ResponsiveRestoreCallback((batch) -> buffer.restoreBatch(batch), this::isActive)
       );
-      storeRegistry = InternalConfigs.loadStoreRegistry(context.appConfigs());
-      storeRegistry.registerStore(registration);
     } catch (InterruptedException | TimeoutException e) {
-      throw new ProcessorStateException("Failed to initialize store.", e);
+      throw new ProcessorStateException("Failed to initialize store " + name(), e);
+    }
+  }
+
+  private boolean isActive() {
+    return context.taskType() == TaskType.ACTIVE;
+  }
+
+  private void initializeBuffer() {
+    buffer.init();
+    final long offset = buffer.offset();
+    registration = new ResponsiveStoreRegistration(
+        name.kafkaName(),
+        topicPartition,
+        offset == -1 ? 0 : offset,
+        buffer::flush
+    );
+    storeRegistry.registerStore(registration);
+    initialized = true;
+    log.info("Initialized buffer");
+  }
+
+  private void maybeTransitionToActive(final String caller) {
+    if (isActive()) {
+      if (!initialized) {
+        log.info("Transitioning from standby to active");
```

Comment:
could we check here that we haven't missed any events? basically the condition should check that the offset in cassandra is the most recent offset in the changelog (on normal read/write path) OR the offset we're currently restoring from (on restore path)